CMakeLists.txt

```diff
@@ -521,6 +521,11 @@ else()
     ENDIF()
   ENDIF()
 
+  find_package(Shapefile) # package libshp-dev
+  if(NOT SHAPEFILE_FOUND)
```

Comment:
We want to make the shapefile dependency optional. For this we need to ensure a few things:

1. We don't try to link against a missing library when it's not there
2. The code that uses shapefile is not compiled when the library is not found.

We get 1. by making the linker flags for shapefiles optional:

```
find_package(Shapefile)
if (SHAPEFILE_FOUND)
  set(MAYBE_SHAPEFILE_LIBRARIES "${SHAPEFILE_LIBRARIES}")
  target_include_directories(UTIL PRIVATE ${LIBSHAPEFILE_INCLUDE_DIR}) 
else()
  set(MAYBE_SHAPEFILE_LIBRARIES "")
endif()
```

This creates a variable that is just empty in case there is no shapefile library found, and only add the header include in case we find the library.

To get 2. we need to tell the C++ code whether it is allowed to use anything that references shapefiles. The typical technique to do that is use defines like `#define FOO_BAR`. Luckily we can pass these options straight to the compiler like:

```
g++ -D FOO_BAR test.cpp -o test
```
Conveniently cmake can generate those flags for us. [Here](https://github.com/Project-OSRM/osrm-backend/blob/master/unit_tests/CMakeLists.txt#L93) is an example of how we do this to pass C++ the location of the test directory.

In our case that would be something like:

```
target_compile_definitions(UTIL PRIVATE COMPILE_DEFINITIONS ENABLE_SHAPEFILE)
```